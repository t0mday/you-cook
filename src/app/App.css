.App {
  text-align: center;
}

.App-header {
  background-image: url('../images/creveta-1327214.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: 60% 15%;
  height: 600px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.banner {
  font-size: 4rem;
  background-color: rgba(250, 235, 215, 0.8);
  margin: 0;
  width: 100%;
  padding: 2rem;
}

h1 {
  color: rgb(83,153,0);
}

h1 span {
  font-style: italic;
}

footer p {
  text-align: right;
  font-style: italic;
  padding: 30px;
  font-family: Helvetica, Tahoma, sans-serif;
}

footer img {
  float: left;
  width: 200px;
  border: none;
}

section {
  border-bottom: 3px solid rgb(17, 54, 112);
}
