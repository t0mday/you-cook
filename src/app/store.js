import { configureStore } from '@reduxjs/toolkit';
import recipesReducer from '../components/recipes/recipesSlice';
import searchReducer from '../components/search/searchSlice';


export const store = configureStore({
  reducer: {
    recipes: recipesReducer,
    search: searchReducer
  }
});
