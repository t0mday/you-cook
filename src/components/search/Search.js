import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { setSearchTerm, clearSearchTerm, selectSearchTerm, addRestriction, removeRestriction, selectRestrictions } from './searchSlice';
import { getRecipes } from '../recipes/recipesSlice';
import './search.css';

const Search = () => {

  const dispatch = useDispatch();
  const searchTerm = useSelector(selectSearchTerm);
  const restrictions = useSelector(selectRestrictions);
  const gf = restrictions.includes('gluten-free');
  const df = restrictions.includes('dairy-free');
  const veggie = restrictions.includes('vegetarian');
  const vegan = restrictions.includes('vegan');

   const handleInput = ({target}) => {
    dispatch(setSearchTerm(target.value));
  };

  const handleCheck = ({target}) => {
    target.checked ? dispatch(addRestriction(target.id)) : dispatch(removeRestriction(target.id));
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    const cleanedSearchTerm = searchTerm.replace(/[&]/g, " ").replace(/[^a-zA-Z ]/g, "");
    dispatch(getRecipes({cleanedSearchTerm, restrictions}));
    dispatch(clearSearchTerm());
  };

  return (
    <div className='test'>
      <h2>Find Recipes</h2>
    <form onSubmit={handleSubmit} >
      <input name="searchField" type="text" onChange={handleInput} value={searchTerm} placeholder='Enter search term food item(s)' required />
      <input id="gluten-free" type="checkbox" checked={gf} onChange={handleCheck} />
      <label htmlFor="gluten-free">Gluten-free</label>
      <input id="dairy-free" type="checkbox" checked={df} onChange={handleCheck} />
      <label htmlFor="dairy-free">Dairy-free</label>
      <input id="vegetarian" type="checkbox" checked={veggie} onChange={handleCheck} />
      <label htmlFor="vegetarian">Vegetarian</label>
      <input id="vegan" type="checkbox" checked={vegan} onChange={handleCheck} />
      <label htmlFor="vegan">Vegan</label>
      <button type="submit" >Search Recipes</button>
    </form>
    </div>
  );
}

export default Search;