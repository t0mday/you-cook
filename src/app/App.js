import React from 'react';
import './App.css';
import attribution from './Edamam_Badge_Transparent.svg';
import Search from '../components/search/Search';
import SearchResults from '../components/searchResults/SearchResults';

function App() {

  return (
    <div className="App">
      <header className="App-header">
        <h1>you Cook</h1>
      </header>
      <section className='favourites'>
        <Search />
      </section>
      <section className='search'>
        {/*<Favourites />*/}
      </section>
      <section className='searchResults'>
        <SearchResults />
      </section>
      <footer>
        <img src={attribution} alt="powered by Edamam" />
      </footer>
    </div>
  );
}

export default App;
