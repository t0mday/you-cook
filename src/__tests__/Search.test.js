import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Search from '../components/search/Search';
import { Provider } from 'react-redux';
import { store } from '../app/store';
import userEvent from '@testing-library/user-event';

const dietaryLabels = [
  'Gluten-free',
  'Dairy-free',
  'Vegetarian',
  'Vegan',
  'Egg-free',
  'Peanut-free',
  'Tree-nut-free',
  'Shellfish-free',
  'Pork-free',
  'Kosher'
];

const cuisineLabels = [
  'Asian',
  'Caribbean',
  'Chinese',
  'East Europe',
  'French',
  'Indian',
  'Italian',
  'Japanese',
  'Mediterranean',
  'Mexican'
];


let asFragment;
beforeEach(() => {
   asFragment = render(<Provider store={store}><Search /></Provider>);
});
// snapshot
test('the Search component should render correctly', () => {
  expect(asFragment).toMatchSnapshot();
});

// tests for behaviour of react controlled inputs

test('the searchbox should be empty to start with, and updated with a user entered value', () => {
  expect(screen.getByPlaceholderText('Enter food to search')).toHaveValue('');
  fireEvent.change(screen.getByPlaceholderText('Enter food to search'), { target: { value: "bangers" } });
  expect(screen.getByPlaceholderText('Enter food to search')).toHaveValue('bangers');
});

test('the cuisine checkboxes should all start unchecked, and each be checked on a user clicking it', () => {
  cuisineLabels.forEach(label => {
  expect(screen.getByLabelText(label)).not.toBeChecked();
  userEvent.click(screen.getByLabelText(label));
  expect(screen.getByLabelText(label)).toBeChecked();
  });
});

test('the dietary restriction checkboxes should all start unchecked, and each be checked on a user clicking it', () => {
  dietaryLabels.forEach(label => {
  expect(screen.getByLabelText(label)).not.toBeChecked();
  userEvent.click(screen.getByLabelText(label));
  expect(screen.getByLabelText(label)).toBeChecked();
  });
});



