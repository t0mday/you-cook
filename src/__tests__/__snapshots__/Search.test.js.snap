// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Search component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <input
          id="searchField"
          name="searchField"
          placeholder="Enter food to search"
          required=""
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Search Recipes
        </button>
        <p
          class="noShowPrompt"
        >
          Scroll down to see results
        </p>
        <div
          class="checkboxes"
        >
          <fieldset
            class="restrictions"
          >
            <legend>
              Restrictions
            </legend>
            <label
              for="gluten-free"
            >
              <input
                id="gluten-free"
                type="checkbox"
              />
              Gluten-free
            </label>
            <label
              for="dairy-free"
            >
              <input
                id="dairy-free"
                type="checkbox"
              />
              Dairy-free
            </label>
            <label
              for="vegetarian"
            >
              <input
                id="vegetarian"
                type="checkbox"
              />
              Vegetarian
            </label>
            <label
              for="vegan"
            >
              <input
                id="vegan"
                type="checkbox"
              />
              Vegan
            </label>
            <label
              for="egg-free"
            >
              <input
                id="egg-free"
                type="checkbox"
              />
              Egg-free
            </label>
            <label
              for="peanut-free"
            >
              <input
                id="peanut-free"
                type="checkbox"
              />
              Peanut-free
            </label>
            <label
              for="tree-nut-free"
            >
              <input
                id="tree-nut-free"
                type="checkbox"
              />
              Tree-nut-free
            </label>
            <label
              for="shellfish-free"
            >
              <input
                id="shellfish-free"
                type="checkbox"
              />
              Shellfish-free
            </label>
            <label
              for="pork-free"
            >
              <input
                id="pork-free"
                type="checkbox"
              />
              Pork-free
            </label>
            <label
              for="kosher"
            >
              <input
                id="kosher"
                type="checkbox"
              />
              Kosher
            </label>
          </fieldset>
          <fieldset
            class="cuisine"
          >
            <legend>
              Cuisine
            </legend>
            <label
              for="Asian"
            >
              <input
                id="Asian"
                type="checkbox"
              />
              Asian
            </label>
            <label
              for="Caribbean"
            >
              <input
                id="Caribbean"
                type="checkbox"
              />
              Caribbean
            </label>
            <label
              for="Chinese"
            >
              <input
                id="Chinese"
                type="checkbox"
              />
              Chinese
            </label>
            <label
              for="Eastern Europe"
            >
              <input
                id="Eastern Europe"
                type="checkbox"
              />
              East Europe
            </label>
            <label
              for="French"
            >
              <input
                id="French"
                type="checkbox"
              />
              French
            </label>
            <label
              for="Indian"
            >
              <input
                id="Indian"
                type="checkbox"
              />
              Indian
            </label>
            <label
              for="Italian"
            >
              <input
                id="Italian"
                type="checkbox"
              />
              Italian
            </label>
            <label
              for="Japanese"
            >
              <input
                id="Japanese"
                type="checkbox"
              />
              Japanese
            </label>
            <label
              for="Mediterranean"
            >
              <input
                id="Mediterranean"
                type="checkbox"
              />
              Mediterranean
            </label>
            <label
              for="Mexican"
            >
              <input
                id="Mexican"
                type="checkbox"
              />
              Mexican
            </label>
          </fieldset>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <input
        id="searchField"
        name="searchField"
        placeholder="Enter food to search"
        required=""
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search Recipes
      </button>
      <p
        class="noShowPrompt"
      >
        Scroll down to see results
      </p>
      <div
        class="checkboxes"
      >
        <fieldset
          class="restrictions"
        >
          <legend>
            Restrictions
          </legend>
          <label
            for="gluten-free"
          >
            <input
              id="gluten-free"
              type="checkbox"
            />
            Gluten-free
          </label>
          <label
            for="dairy-free"
          >
            <input
              id="dairy-free"
              type="checkbox"
            />
            Dairy-free
          </label>
          <label
            for="vegetarian"
          >
            <input
              id="vegetarian"
              type="checkbox"
            />
            Vegetarian
          </label>
          <label
            for="vegan"
          >
            <input
              id="vegan"
              type="checkbox"
            />
            Vegan
          </label>
          <label
            for="egg-free"
          >
            <input
              id="egg-free"
              type="checkbox"
            />
            Egg-free
          </label>
          <label
            for="peanut-free"
          >
            <input
              id="peanut-free"
              type="checkbox"
            />
            Peanut-free
          </label>
          <label
            for="tree-nut-free"
          >
            <input
              id="tree-nut-free"
              type="checkbox"
            />
            Tree-nut-free
          </label>
          <label
            for="shellfish-free"
          >
            <input
              id="shellfish-free"
              type="checkbox"
            />
            Shellfish-free
          </label>
          <label
            for="pork-free"
          >
            <input
              id="pork-free"
              type="checkbox"
            />
            Pork-free
          </label>
          <label
            for="kosher"
          >
            <input
              id="kosher"
              type="checkbox"
            />
            Kosher
          </label>
        </fieldset>
        <fieldset
          class="cuisine"
        >
          <legend>
            Cuisine
          </legend>
          <label
            for="Asian"
          >
            <input
              id="Asian"
              type="checkbox"
            />
            Asian
          </label>
          <label
            for="Caribbean"
          >
            <input
              id="Caribbean"
              type="checkbox"
            />
            Caribbean
          </label>
          <label
            for="Chinese"
          >
            <input
              id="Chinese"
              type="checkbox"
            />
            Chinese
          </label>
          <label
            for="Eastern Europe"
          >
            <input
              id="Eastern Europe"
              type="checkbox"
            />
            East Europe
          </label>
          <label
            for="French"
          >
            <input
              id="French"
              type="checkbox"
            />
            French
          </label>
          <label
            for="Indian"
          >
            <input
              id="Indian"
              type="checkbox"
            />
            Indian
          </label>
          <label
            for="Italian"
          >
            <input
              id="Italian"
              type="checkbox"
            />
            Italian
          </label>
          <label
            for="Japanese"
          >
            <input
              id="Japanese"
              type="checkbox"
            />
            Japanese
          </label>
          <label
            for="Mediterranean"
          >
            <input
              id="Mediterranean"
              type="checkbox"
            />
            Mediterranean
          </label>
          <label
            for="Mexican"
          >
            <input
              id="Mexican"
              type="checkbox"
            />
            Mexican
          </label>
        </fieldset>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
